File: Arrays&Hashing/valid_anagram/valid-anagram.ts
function isAnagram(s: string, t: string): boolean {
    if (s.length !== t.length)
        return false;

    const arrayS = s.split("");
    const arrayT = t.split("");

    arrayS.sort();
    arrayT.sort();

    return JSON.stringify(arrayS) === JSON.stringify(arrayT);
}

const test1Word1 = "anagram";
const test1Word2 = "nagaram";

console.log("Test 1: ", isAnagram(test1Word1, test1Word2));

const test2Word1 = "rat";
const test2Word2 = "car";

console.log("Test 2: ", isAnagram(test2Word1, test2Word2));

File: src/main/java/arrays_hashing/ValidAnagram.java
package arrays_hashing;

import java.util.Arrays;

public class ValidAnagram {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length())
            return false;

        char[] arrayS = s.toCharArray();
        char[] arrayT = t.toCharArray();

        Arrays.sort(arrayS);
        Arrays.sort(arrayT);

        return Arrays.equals(arrayS, arrayT);
    }

    public static void main(String[] args) {
        ValidAnagram solution = new ValidAnagram();

        String test1Word1 = "anagram";
        String test1Word2 = "nagaram";

        System.out.println("Test 1: " + solution.isAnagram(test1Word1, test1Word2));

        String test2Word1 = "rat";
        String test2Word2 = "car";

        System.out.println("Test 2: " + solution.isAnagram(test2Word1, test2Word2));
    }
}
